src/Orleans/Lifecycle/ILifecycleObserver.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Lifecycle observer used to handle start and stop notification.
+    /// </summary>
+    public interface ILifecycleObserver
+    {
+        /// <summary>
+        /// Handle start notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStart(CancellationTokenSource cts = null);
```

Comment:
What's the purpose of a `CancellationTokenSource` here? I would expect a `CancellationToken`